Sync/Synchronizers/TabsSynchronizer.swift

```diff
@@ -20,6 +20,78 @@ public class TabsSynchronizer: BaseSingleCollectionSynchronizer, Synchronizer {
         return TabsStorageVersion
     }
 
+    var tabsRecordLastUpload: Timestamp {
+        set(value) {
+            self.prefs.setLong(value, forKey: "lastTabsUpload")
+        }
+
+        get {
+            return self.prefs.unsignedLongForKey("lastTabsUpload") ?? 0
+        }
+    }
+
+    private func createOwnTabsRecord(tabs: [RemoteTab]) -> Record<TabsPayload> {
+        let guid = self.scratchpad.clientGUID
+
+        let jsonTabs: [JSON] = optFilter(tabs.map { $0.toJSON() })
+        let tabsJSON = JSON([
+            "id": guid,
+            "clientName": self.scratchpad.clientName,
+            "tabs": jsonTabs
+        ])
+        log.debug("Sending tabs JSON \(tabsJSON.toString(pretty: true))")
+        let payload = TabsPayload(tabsJSON)
+        return Record(id: guid, payload: payload, ttl: ThreeWeeksInSeconds)
+    }
+
+    // annoyingly can't use direct comparison (==) as our tabs come from the DB with a null client ID as opposed to the remote server who sends us our remote client ID
+    // TODO: there must be a better way to do this but brain freeze right now
+    private func tabRecordsAreEqual(localTabs: [RemoteTab], remoteTabs: [RemoteTab]) -> Bool {
```

Comment:
Shouldn't need to do this. We're the ones who uploaded the existing record! We know exactly what we put in it!